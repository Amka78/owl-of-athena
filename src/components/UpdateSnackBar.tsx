import React, { useEffect, FunctionComponent } from "react";
import { Snackbar } from "react-native-paper";
import { Message, MessageKeys } from "../constants";

export const UpdateSnackBar: FunctionComponent = () => {
    const [showReload, setShowReload] = React.useState(false);
    const [
        waitingWorker,
        setWaitingWorker,
    ] = React.useState<ServiceWorker | null>(null);

    useEffect(() => {
        // @ts-ignore
        navigator.serviceWorker
            .register("/expo-service-worker.js", {
                scope: "/",
            })
            .then((registration: ServiceWorkerRegistration) => {
                registration.onupdatefound = (): void => {
                    setShowReload(true);
                    setWaitingWorker(registration.waiting);
                };
                console.debug(registration);
            })
            .catch((error: any) => {
                console.log(error);
            });
    }, []);

    const reloadPage = (): void => {
        // @ts-ignore
        waitingWorker?.postMessage({ type: "SKIP_WAITING" });
        setShowReload(false);
        // @ts-ignore
        window.location.reload(true);
    };

    return (
        <Snackbar
            visible={showReload}
            onDismiss={reloadPage}
            action={{
                label: Message.get(MessageKeys.update_snack_bar_action_label),
                onPress: reloadPage,
            }}
        >
            {Message.get(MessageKeys.update_snack_bar_title)}
        </Snackbar>
    );
};
