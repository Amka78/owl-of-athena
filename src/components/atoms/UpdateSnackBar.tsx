//#region Import modules
import React, { FunctionComponent } from "react";
import { Snackbar } from "react-native-paper";

import { Message, MessageKeys } from "../../constants";
import { useAppUpdate } from "../../hooks";
//#endregion

//#region Types
export type UpdateSnackBarProps = {
    showReload: boolean;
    reloadPageCallback?: () => void;
};
//#endregion

//#region Components
export const UpdateSnackBarCore: FunctionComponent<UpdateSnackBarProps> = (
    props: UpdateSnackBarProps
) => {
    return (
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        //@ts-ignore
        <Snackbar
            visible={props.showReload}
            onDismiss={props.reloadPageCallback}
            action={{
                label: Message.get(MessageKeys.update_snack_bar_action_label),
                onPress: props.reloadPageCallback,
            }}
        >
            {Message.get(MessageKeys.update_snack_bar_title)}
        </Snackbar>
    );
};

export const UpdateSnackBar: FunctionComponent = () => {
    const appUpdateHook = useAppUpdate();
    return <UpdateSnackBarCore {...appUpdateHook}></UpdateSnackBarCore>;
};
//#endregion
