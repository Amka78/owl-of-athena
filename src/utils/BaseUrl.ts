import Constants from "expo-constants";

/**
 * Manage API-URL for each environment.
 *
 * @export
 * @class BaseUrl
 */
export default class BaseUrl {
    public static get(relCh?: string) {
        if (!relCh && Constants.manifest.releaseChannel) {
            relCh = Constants.manifest.releaseChannel;
        }

        if (relCh !== undefined) {
            if (relCh!.includes("staging")) {
                return IWinksRestAPI.staging.url;
            } else if (relCh!.includes("prod")) {
                return IWinksRestAPI.prod.url;
            }
        }

        return IWinksRestAPI.dev.url;
    }

    public static getDownloadUrl(
        image: string | string[],
        index?: number
    ): string | undefined {
        if (typeof image === "undefined") {
            return undefined;
        } else if (typeof image === "string") {
            return BaseUrl.get() + image;
        } else {
            if (index === undefined) {
                index = 0;
            }
            return BaseUrl.get() + image![index];
        }
    }
}

export const IWinksRestAPI = {
    dev: {
        url: "https://api.iwinks.io/"
    },
    staging: {
        url: "https://api.iwinks.io/"
    },
    prod: {
        url: "https://api.iwinks.io/"
    }
};
