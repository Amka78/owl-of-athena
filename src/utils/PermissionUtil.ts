import * as Permissions from "expo-permissions";
import { Linking } from "react-native";

export default class PermissionUtil {
    public static async getPermission(permission: any) {
        const { status } = await Permissions.askAsync(permission);
        if (status !== "granted") {
            Linking.openURL("app-settings:");
            return false;
        }
        return true;
    }
}
