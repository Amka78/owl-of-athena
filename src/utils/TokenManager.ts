import { Platform } from "@unimodules/core";
import * as SecureStore from "expo-secure-store";
import { AsyncStorage } from "react-native";

enum SecureStoreKeys {
    JWT_TOKEN = "jwt_token"
}
/**
 * Token Management Class
 *
 * @export
 * @class TokenManager
 */
export default class TokenManager {
    public static async hasToken(): Promise<boolean> {
        const result = await this.getPerPlatform(SecureStoreKeys.JWT_TOKEN);

        return result !== undefined;
    }

    /**
     * get token
     *
     * @static
     * @returns {(Promise<string | undefined>)}
     * @memberof TokenManager
     */
    public static async get(): Promise<string | undefined> {
        const token = await this.getPerPlatform(SecureStoreKeys.JWT_TOKEN);

        if (token) {
            return token;
        }

        return undefined;
    }

    /**
     * set token
     *
     * @static
     * @param {TokenDto} value
     * @memberof TokenManager
     */
    public static async set(value: string) {
        await this.setPerPlatform(SecureStoreKeys.JWT_TOKEN, value);
    }

    public static async reset() {
        await this.deletePerPlatform(SecureStoreKeys.JWT_TOKEN);
    }

    private static async getPerPlatform(key: string): Promise<string | null> {
        if (Platform.OS === "web") {
            return await AsyncStorage.getItem(key);
        }

        return await SecureStore.getItemAsync(key);
    }

    private static async setPerPlatform(
        key: string,
        value: string
    ): Promise<void> {
        if (Platform.OS === "web") {
            return await AsyncStorage.setItem(key, value);
        }

        return await SecureStore.setItemAsync(key, value);
    }

    private static async deletePerPlatform(key: string): Promise<void> {
        if (Platform.OS === "web") {
            return await AsyncStorage.removeItem(key);
        }

        return await SecureStore.deleteItemAsync(key);
    }
}
