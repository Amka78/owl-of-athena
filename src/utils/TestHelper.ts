import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
