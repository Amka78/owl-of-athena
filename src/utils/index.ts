import BaseUrl from "./BaseUrl";
export { BaseUrl };
