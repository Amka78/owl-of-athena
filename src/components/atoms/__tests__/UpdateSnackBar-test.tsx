//#region Import Modules
import "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";

import { createMock, toJson } from "../../../utils/TestHelper";
import { UpdateSnackBar, UpdateSnackBarProps } from "../UpdateSnackBar";
//#endregion

//#region Test
let component: ShallowWrapper<UpdateSnackBarProps, unknown, unknown>;
describe("UpdateSnackBar UnitTest", () => {
    it("renders correctly", () => {
        component = createMock(<UpdateSnackBar></UpdateSnackBar>);

        expect(toJson(component)).toMatchSnapshot();
    });
});
//#endregion
