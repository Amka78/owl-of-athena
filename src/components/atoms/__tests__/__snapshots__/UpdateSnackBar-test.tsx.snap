// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateSnackBar UnitTest renders correctly 1`] = `
<Component
  reloadPageCallback={[Function]}
  showReload={false}
/>
`;
